clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2429,10 +2432,12 @@ private int internalProcessConsumerRecord(
         sizeOfPersistedData =
             processKafkaDataMessage(consumerRecord, partitionConsumptionState, leaderProducedRecordContext);
       }
+      long processingFinishedTimeNs = System.nanoTime();
```

Comment:
Curious, do we really need `nanoTime` ? can we get away with millis as it will convert to millis later.

Reply:
Because we compare it to `beforeProcessingRecordTimestampNs` which is in nanos.

Reply:
But do we need nanotime ? I tried a mini benchmark, different OS has different perf on those, but `currenttimemillis` is faster than `nanotime` on mac. 

nanoTime:
Benchmark                                                       Mode  Cnt   Score    Error   Units
VeniceClientBenchmark.runAvroClientQueries                               avgt    4  106.142 ±   6.660   ns/op


millis:
Benchmark                                                       Mode  Cnt   Score    Error   Units
VeniceClientBenchmark.runAvroClientQueries                      avgt    6  70.159 ±  0.185   ns/op

Reply:
So... I refreshed my memory on nano time and I think there are issues in this PR...

First of all, IIUC, nano time is consistent with itself but is not necessarily the correct time. So it's not correct to just divide it by 1M and use it as if it were the currentTimeMillis, as I do here in a few places.

Secondly, it seems like nano time is not safe to use across different threads. I guess it's based on ticks of each core...? If that is true, then it wouldn't be correct to measure nano time in one thread and compare against the nano time of another thread, which may be what we do with this particular variable. In which case, might be preferable to change it to millis for correctness sake, like you suggested.

Reply:
I had a bug indeed, and I fixed it (verified by unit test) in the last commit.

I would prefer to keep this a nano time though, as that is the recommended approach for timing the code. The thing about nano time being different on different cores is outdated, that is not true anymore since Java 7.